src/database/mod.rs

```diff
@@ -1,4 +1,8 @@
-#![allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
+#![allow(
+    clippy::cast_sign_loss,
+    clippy::cast_possible_truncation,
+    clippy::needless_range_loop
```

Comment:
I agree with clippy here, there are needless range loops in this file and its suggestion is better style.